### ISSUE TITLE
Python Backend "triton_python_backend_utils" lost many attributes from version 22.05 to 22.06 causing models to fail 

### ISSUE THREAD

**Issue by tobiasvitt** [triton-inference-server/server#6535]
**Description**
We are using triton server 22.05-py3 and don't face any issues with the following code inside our python models:

```
import triton_python_backend_utils as pb_utils
class TritonPythonModel:
    ...
    def execute(self, requests: List[List[pb_utils.Tensor]]) -> List[List[pb_utils.Tensor]]:
        ...
```

However, we when update the triton server to 22.06-py3, we get the following error:

`AttributeError: module 'triton_python_backend_utils' has no attribute 'Tensor'
`
The same thing is also true for other Attributes like `pb_utils.InferenceRequest`

There are many reports from other users that face the same issue [here](https://github.com/triton-inference-server/server/issues/4743).

**Triton Information**
What version of Triton are you using?
22.05 works, 22.06 not

Are you using the Triton container or did you build it yourself?
nvcr.io/nvidia/tritonserver:22.05-py3 works, nvcr.io/nvidia/tritonserver:22.06-py3 not

**To Reproduce**
Steps to reproduce the behavior.
- Create a python backend model with similar type hints/annotations as above. No custom python env
- Load the model in triton 22.05, it will work
- Load the model in triton 22.06, it will fail

**Expected behavior**
Backward compatibility for attributes like Tensor and InferenceRequest.

**Comment by krishung5** [comment]
Hi @tobiasvitt, thanks for filing the report. This is a known issue with type annotations and we are working on the fix. A quick WAR that might unblock you is to modify the model to import `c_python_backend_utils ` like below:
```
import c_python_backend_utils as c_utils
...
def execute(self, requests: List[c_utils.InferenceRequest]):
  ...
```

**Comment by tobiasvitt** [comment]
Hi @krishung5, do you have any estimate of when a fix will be available? I'm just asking to know if we can postpone the version upgrade or should adapt all our models to `c_python_backend_utils`.

**Comment by TopCoder2K** [comment]
Same problem with `nvcr.io/nvidia/tritonserver:23.12-py3` :( Although the [tutorials](https://github.com/triton-inference-server/python_backend?tab=readme-ov-file#error-handling) confidently use `triton_python_backend_utils.InferenceResponse`, I get
```
I1223 17:27:01.401467 19 pb_stub.cc:278]  Failed to initialize Python stub for auto-complete: AttributeError: module 'triton_python_backend_utils' has no attribute 'InferenceResponse'
```

@krishung5, is the problem still not fixed?

**Comment by zzk2021** [comment]
24.06 also has problem
